SolmangoCLI/Statics/CommandsHandler.cs

```diff
@@ -55,31 +61,154 @@ public static async Task ScrapeCommand(ArgumentsHandler handler, IServiceProvide
-        var rentExemptionAmmount = await activeRpcClient.GetMinimumBalanceForRentExemptionAsync(TokenProgram.TokenAccountDataSize);
+        var connectionOption = services.GetRequiredService<IOptionsMonitor<ConnectionSettings>>();
+        var rpcClient = ClientFactory.GetClient(connectionOption.CurrentValue.ClusterEndpoint);
+        try
+        {
+            var hash = File.ReadAllText(handler.GetPositional(0));
+            var hashList = JsonConvert.DeserializeObject<ImmutableList<string>>(hash);
+            if (hashList != null && hashList.Count > 0)
+            {
+                ConsoleProgressBar progressBar = new ConsoleProgressBar(50);
+                var res = await SolmangoNET.Solmango.GetOwnersByCollection(rpcClient, hashList, progressBar);
+                if (res.TryPickT1(out var ex, out var owners))
+                {
+                    logger.LogError(ex.Reason);
+                    progressBar.Dispose();
+                    return false;
+                }
+                progressBar.Dispose();
+                var json = JsonConvert.SerializeObject(owners, Formatting.Indented);
+                File.WriteAllText(handler.GetPositional(1), json);
+                int sum = 0;
+                foreach (var pair in owners)
+                {
+                    sum += pair.Value.Count;
+                }
+
+                logger.LogInformation("Found {holders} holders after performing scraping on {sum} mints ", owners.Keys.Count, sum);
+                return true;
+            }
+            else
+            {
+                logger.LogError("Couldn't find the hash list path or the file is empty");
+                return false;
+            }
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex.Message);
+            return false;
+        }
+    }
 
-        var lortAccounts = await GetOwnedTokenAccounts(toWalletAccount, tokenMint, "", activeRpcClient);
+    public static async Task<bool> DistributeTokens(ArgumentsHandler handler, IServiceProvider services, ILogger logger)
+    {
+        var connectionOption = services.GetRequiredService<IOptionsMonitor<ConnectionSettings>>();
+        var rpcClient = ClientFactory.GetClient(connectionOption.CurrentValue.ClusterEndpoint);
+        ConsoleProgressBar progressBar = new ConsoleProgressBar(50);
+        int sum = 0;
+        List<string> failedAddresses = new List<string>();
+        try
+        {
+            var key = File.ReadAllText(handler.GetPositional(0));
+            //gets keys
+            var keys = JsonConvert.DeserializeObject<KeyPair>(key);
+            Account sender = new Account(keys!.PrivateKey, keys.PublicKey);
+            //gets dictionary
+            var str = File.ReadAllText(handler.GetPositional(2));
+            var dic = JsonConvert.DeserializeObject<Dictionary<string, List<string>>>(str);
+            int i = 1;
+            foreach (var pair in dic!)
```

Comment:
@Nickmadia do not force `dic` variable to be non null, check if null before using it instead.